cataclysmbnteam/Cataclysm-BN
# New helicopter based scenario and fix for inability to set helicopter as starting vehicle

New helicopter based scenario where player starts with helicopter avalible.

"You manage to get to the chopper right when Cataclysm begins.  But are there safe places to land left in this world now?"
"You manage to make it to your UH-60. Your co-pilot didn't."

In this scenario player starts in wilderness but with military helicopter available. For now there is military pilot proffesion with UH-60 as starting vehicle. This scenario has no starting NPC.
Scenario cost 1 point. But only avalible proffesion costs 7 points.



#### Summary

SUMMARY: Content "New helicopter based scenario"
New wilderness scenario where player starts with helicopter available right from the start.



#### Purpose of change
New scenario designed to provide helicopter experience to player right from the start and get new source of feedback on helicopters.



#### Describe the solution
1) Fix bug that does not allow to properly set helicopter as starting vehicle. Problem was in rule that selects location of vehicle to spawn- it does not took in accout helicopters properly. 
2) Add scenario with helicopter as starting vehicle.
3) Add new proffesion "UH-60 Blackhawk pilot" with UH-60 Blackhawk as starting vehicle.


#### Describe alternatives you've considered



#### Testing 
1) Start new "Flight of the Valkyries" scenario
2) Check if helicopter is avalible.


#### Additional context
Helicopters sometime can spawn somewhat far from player. But it is how starting vehicle spawn works now. Still player getting location of the vehicle on the map.
**Found workaround- try to spawn helicopter in the filed first - this way it spawns much closer.**

Note 1: Helicopter spawns with deley for few turn. Probalby need issue on that.
Note 2:  Spawning Apache producing error. Need issue on that.
Note 3: I've set priority to find location to vehicle spawn starting vehicle to fields instead of roads, especially for wilderness starts. Usually it should provide much closer position for starting vehicle since fields are everywhere but roads are not.
Still potentially it could move vehicle further away with city starting location, but it usually should not be that far anyway- since, again, fields are everywhere.
